src/VisualStudio/Core/Def/Implementation/EditAndContinue/VisualStudioActiveStatementProvider.cs

```diff
@@ -30,16 +30,14 @@ public Task<ImmutableArray<ActiveStatementDebugInfo>> GetActiveStatementsAsync(C
                 // TODO: return empty outside of debug session.
                 // https://github.com/dotnet/roslyn/issues/24325
 
-                int unexpectedError = 0;
                 var completion = new TaskCompletionSource<ImmutableArray<ActiveStatementDebugInfo>>();
                 var builders = default(ArrayBuilder<ArrayBuilder<ActiveStatementDebugInfo>>);
                 int pendingRuntimes = 0;
                 int runtimeCount = 0;
 
-                var workList = DkmWorkList.Create(CompletionRoutine: _ =>
-                {
-                    completion.TrySetException(new InvalidOperationException($"Unexpected error enumerating active statements: 0x{unexpectedError:X8}"));
-                });
+                // No exception should be thrown in case of errors on the debugger side. 
+                // The debugger is responsible to provide telemetry for error cases.
+                var workList = DkmWorkList.Create(CompletionRoutine: _ => { }); 
```

Comment:
What happens now if there's any logic error in our code? Will nothing complete the task and as a result we'll hang or deadlock or something?

Reply:
I believe we will just ignore the error on this level. The debugger will handle the error, we will just decide if we can proceed with the statement or skip it. @r-ramesh to confirm.

---
In reply to: [197511202](https://github.com/dotnet/roslyn/pull/28070#discussion_r197511202) [](ancestors = 197511202)

Reply:
DkmWorkList.Create can take null as the parameter (no completion routine).

Reply:
public static DkmWorkList Create(DkmWorkListCompletionRoutine CompletionRoutine);

---
In reply to: [198272533](https://github.com/dotnet/roslyn/pull/28070#discussion_r198272533) [](ancestors = 198272533)

Reply:
this is the only signature for the method.

---
In reply to: [198310766](https://github.com/dotnet/roslyn/pull/28070#discussion_r198310766) [](ancestors = 198310766,198272533)